MDANSE/Src/MDANSE/Mathematics/Arithmetic.py

```diff
@@ -125,20 +124,20 @@ def weighted_sum(
     ----------
     values : Dict[str, np.ndarray]
         Dictionary of data arrays containing analysis results.
-    weights : Dict[str, float]
-        Dictionary of scaling factors per dataset
-    key : str
-        A string data set name with formatting elements (placeholders for chemical element labels)
+    match_key: str
+        A key used to generate the list of matches to sum over.
+    match_labels: Iterable
+        The labels used to generate the list of matches to sum over.
 
     Returns
     -------
     np.ndarray
-        total sum of all the component arrays scaled by their weights
+        Total sum of all the component arrays scaled by their weights
     """
-    weightedSum = 0.0
-    matches = {key % k for k in weights if k not in ["sum"]}
+    matches = [match_key % val[0] for val in match_labels]
```

Comment:
Set contains should be faster.
```suggestion
    matches = {match_key % val[0] for val in match_labels}
```